packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart

```diff
@@ -537,14 +577,143 @@ class _ListWheelScrollViewState extends State<ListWheelScrollView> {
             clipToSize: widget.clipToSize,
             renderChildrenOutsideViewport: widget.renderChildrenOutsideViewport,
             offset: offset,
-            children: widget.children,
+            childDelegate: widget.childDelegate,
           );
         },
       ),
     );
   }
 }
 
+/// Element that supports building children lazily for [ListWheelViewport].
+class ListWheelElement extends RenderObjectElement {
+  /// Creates an element that lazily builds children for the given widget.
+  ListWheelElement(ListWheelViewport widget) : super(widget);
+
+  @override
+  ListWheelViewport get widget => super.widget;
+
+  @override
+  RenderListWheelViewport get renderObject => super.renderObject;
+
+  /// A cache of widgets so that we don't have to rebuild every time.
+  final Map<int, Widget> _childWidgets = new HashMap<int, Widget>();
+
+  /// The map containing all active child elements.
+  final SplayTreeMap<int, Element> _childElements = new SplayTreeMap<int, Element>();
+
+  @override
+  void update(covariant ListWheelViewport newWidget) {
+    final ListWheelViewport oldWidget = widget;
+    super.update(newWidget);
+    final SliverChildDelegate newDelegate = newWidget.childDelegate;
+    final SliverChildDelegate oldDelegate = oldWidget.childDelegate;
+    if (newDelegate != oldDelegate &&
+        (newDelegate.runtimeType != oldDelegate.runtimeType || newDelegate.shouldRebuild(oldDelegate)))
+      performRebuild();
+  }
+
+  /// The number of children that will be provided. Be null if number of
+  /// children is infinite.
+  int get childCount => widget.childDelegate.estimatedChildCount;
+
+  @override
+  void performRebuild() {
+    _childWidgets.clear();
+    super.performRebuild();
+    if(_childElements.isEmpty)
+      return;
+
+    final int firstIndex = _childElements.firstKey();
+    final int lastIndex = _childElements.lastKey();
+
+    for (int index = firstIndex; index <= lastIndex; ++index) {
+      final Element newChild = updateChild(_childElements[index], _build(index), index);
+      if (newChild != null) {
+        _childElements[index] = newChild;
```

Comment:
Note to self: these calls to _childElements operator[] will have O(log(n)) amortized time, so the loop will be O(n*log(n)).

In other words, for adding/removing single elements, this is faster than using a List for storage, but for iterating over n elements, it will be slower by a factor of log(n).

In the worst case, we will have to re-add or remove N elements to/from the map.  We should consider an approach to simplify this into two loops of complexity O(n): one to update the elements in the tree, and another pass to re-populate the map and keep it up to date.

Reply:
I'm still confuse here about how you want to optimize this. Can you explain further?

Reply:
I'm not sure yet.  With the current implementation, we are guaranteed a linearithmic (O(NlogN) worst-case complexity all the time.  By switching to another implementation, such as using a list, we might get linear performance in the optimal case, but we could also end up with quadratic performance in the worst case.